tools/check-guide/src/main.rs

```diff
@@ -0,0 +1,113 @@
+//! This crate verifies the correctness of every Hermes command in the guide by:
+//! 1. Extracting every line in the guide with '{{#template *templates/commands/hermes*}}', a macro call for mdbook template.
+//! 2. Replace every template call with the content of the template. It will replace macro call by what should be an Hermes command.
+//! 3. Check that an `EntryPoint` can be created from the command.
+
+use clap::Parser;
+use ibc_relayer_cli::entry::EntryPoint;
+use lazy_static::lazy_static;
+use mdbook_template::replace_template;
+use mdbook_template::utils::SystemFileReader;
+use regex::Regex;
+use std::{
+    collections::HashSet,
+    ffi::OsStr,
+    fs::File,
+    io::{BufRead, BufReader},
+    path::{Path, PathBuf},
+};
+use walkdir::WalkDir;
+
+#[derive(Debug)]
+enum ParseError {
+    IncorrectHermesCommand(clap::Error),
+}
+
+impl From<clap::Error> for ParseError {
+    fn from(e: clap::Error) -> Self {
+        ParseError::IncorrectHermesCommand(e)
+    }
+}
+
+// Constants
+lazy_static! {
+    // Path to the guide.
+    static ref GUIDE_PATH: PathBuf = PathBuf::from("guide/src/");
+
+    // Path to the templates folder where hermes commands templates are defined.
+    static ref TEMPLATES_PATH: PathBuf = PathBuf::from("guide/src/templates/commands/hermes/");
+
+    // List of directories which should not be visited when checking for correctness.
+    static ref EXCLUSIONS: HashSet<&'static str> = HashSet::from(["templates", "assets", "images", "theme"]);
+
+    // Regex to match macro calls in the guide
+    static ref TEMPLATE_RE: Regex = Regex::new(r"\{{2}#template.*templates/commands/hermes/.*\}\}").unwrap();
+
+    // FileReader for mdbook_template
+    static ref FILEREADER: SystemFileReader = SystemFileReader::default();
+}
+
+fn check_correctness<'a, T>(command: T) -> Result<(), ParseError>
+where
+    T: IntoIterator<Item = &'a str>,
+    // Returns an error if the command cannot be parsed.
+{
+    EntryPoint::try_parse_from(command)?;
+    Ok(())
+}
+
+fn verify_line(line: &str, path: &Path, line_number: i32) -> i32 {
+    // Extracts macro calls from a command, replace them with the content of the template and check that the command is correct.
+    // Returns the number of errors encountered.
```

Comment:
Let's turn these into doc comments for the function.
```suggestion
/// Extracts macro calls from a command, replace them with the content of the template and check that the command is correct.
/// Returns the number of errors encountered.
fn verify_line(line: &str, path: &Path, line_number: i32) -> i32 {
```